feat: add better docs for timestamp.to_{hour,day} (#3234)
getsentry/sentry-docs: src/docs/product/sentry-basics/guides/discover/index.mdx

![Context menu](013.png)

### Errors by Hour or Day

> The `timestamp.to_hour` and `timestamp.to_day` fields round down the timestamp to the last UTC hour or day. For example, an event with a `timestamp` of Mar 12, 2021 2:26:50 PM UTC will have a `timestamp.to_hour` of Mar 12, 2021 2:00:00 PM UTC and `timestamp.to_day` of Mar 12, 2021 12:00:00 AM UTC. This allows you to roll up events: for example, you can show hourly hit counts for errors hit at least twice per hour:

Reviewer: ```suggestion
The `timestamp.to_hour` and `timestamp.to_day` fields round down the timestamp to the last UTC hour or day respectively. For example, an event with a `timestamp` value of Mar 12, 2021 2:26:50 PM UTC, will have a `timestamp.to_hour` value of Mar 12, 2021 2:00:00 PM UTC and `timestamp.to_day` value of Mar 12, 2021 12:00:00 AM UTC. This allows you to roll up events. For example, you can show hourly hit counts for errors hit at least twice per hour:
```

Author: applied change sans comma. thank you for the suggestion.

> The `timestamp.to_hour` and `timestamp.to_day` fields round down the timestamp to the last UTC hour or day respectively. For example, an event with a `timestamp` value of Mar 12, 2021 2:26:50 PM UTC will have a `timestamp.to_hour` value of Mar 12, 2021 2:00:00 PM UTC and `timestamp.to_day` value of Mar 12, 2021 12:00:00 AM UTC. This allows you to roll up events. For example, you can show hourly hit counts for errors hit at least twice per hour:

Reviewer 2: This shouldn't just always use UTC time. Like the `timestamp` field, this should respect the user's timezone preference.

Author: There's two parts here: calculation and formatting. For the calculation part, it's complicated to pass through the right time zone value, and raises questions about sharing queries. For the formatting part, that was broken on prod due to setting timezone to `Universal` instead of `UTC`, and was fixed in getsentry/sentry#24443.

> The `timestamp.to_hour` and `timestamp.to_day` fields round down the timestamp to the last UTC hour or day respectively. For example, an event with a `timestamp` value of Mar 12, 2021 2:26:50 PM UTC will have a `timestamp.to_hour` value of Mar 12, 2021 2:00:00 PM UTC and `timestamp.to_day` value of Mar 12, 2021 12:00:00 AM UTC. This allows you to roll up events. For example, you can show hourly hit counts for errors hit at least twice per hour:

Reviewer 3: ```suggestion
The `timestamp.to_hour` and `timestamp.to_day` fields round down the timestamp to the last UTC hour or day respectively. For example, an event with a `timestamp` value of Mar 12, 2021 2:26:50 PM UTC will have a `timestamp.to_hour` value of Mar 12, 2021 2:00:00 PM UTC and `timestamp.to_day` value of Mar 12, 2021 12:00:00 AM UTC. This allows you to group events. For example, you can show hourly hit counts for errors hit at least twice per hour:
```

![Context menu](013.png)

### Errors by Hour or Day

The `timestamp.to_hour` and `timestamp.to_day` fields round down the timestamp to the last hour or day respectively. For example, an event with a `timestamp` value of Mar 12, 2021 2:26:50 PM UTC will have a `timestamp.to_hour` value of Mar 12, 2021 2:00:00 PM UTC and `timestamp.to_day` value of Mar 12, 2021 12:00:00 AM UTC. This allows you to group events. For example, you can show hourly hit counts for errors hit at least twice per hour:

![Hourly counts for errors hit at least twice](014.png)

> All timestamps will be displayed in your preferred timezone according to your user settings in **User Settings > Account > Account Details**.

Reviewer 4: ```suggestion
All timestamps are displayed in your preferred timezone based on your user settings in **User Settings > Account > Account Details**.
```
